src/Microsoft.AspNetCore.Razor.VSCode/src/RazorLanguageServerClient.ts

```diff
@@ -90,22 +96,42 @@ export class RazorLanguageServerClient implements vscode.Disposable {
     }
 
     public async start() {
+        if (this.startHandle) {
+            return this.startHandle;
+        }
+
+        let resolve: () => void = Function;
+        let reject: () => void = Function;
+        this.startHandle = new Promise<void>((resolver, rejecter) => {
+            resolve = resolver;
+            reject = rejecter;
+        });
+
         try {
             this.logger.logMessage('Starting Razor Language Server...');
             this.startDisposable = await this.client.start();
             this.logger.logMessage('Server started, waiting for client to be ready...');
             await this.client.onReady();
-            this.logger.logMessage('Server started and ready!');
 
             this.isStarted = true;
+            this.logger.logMessage('Server started and ready!');
             this.eventBus.emit(events.ServerStart);
+
+            for (const listener of this.onStartedListeners) {
+                await listener();
+            }
+
```

Comment:
Not essential, but we *could* empty `this.onStartedListeners` here so we're not preventing them and their chain of references from being GCed.

Reply:
Eh, they shouldn't be super significant and disposing of the server client object should be sufficient enough to allow for GC.